### PR TITLE
APIGOV-17716 - unify logger for discovery and traceability agents

### PR DESCRIPTION
**Pull request by jcollins-axway**

### CODE REVIEW

**Review by sbolosan — APPROVED** [review]

**Review by kflan-io — APPROVED** [review]

File: pkg/util/log/log.go [hunk 1]
```diff
@@ -1,57 +1,123 @@
...
+	"github.com/elastic/beats/v7/libbeat/logp"
 	"github.com/sirupsen/logrus"
 )
 
+const (
+	debugSelector = "apic-agents"
+	traceSelector = "apic-agents-trace"
+)
+
+var isLogP bool
```
Comment (dfeldick) [inline pkg/util/log/log.go:15]: Could have maybe set a default value of either T or F. Then only need to call fn to change value if so desired.

File: pkg/util/log/log.go [hunk 2]
```diff
@@ -1,57 +1,123 @@
...
+
+//SetIsLogP -
+func SetIsLogP() {
```
Comment (dfeldick) [inline pkg/util/log/log.go:18]: Could just have a single fn SetIsLogP(bool). Don't know which is better. In fact, in combination with a default value, you probably never need to even have one of the functions. If the default is true, you might only ever need to set it to false. Or if the default is false, you might only ever need to set it to true.

**Review by sbolosan — APPROVED** [review]

File: pkg/util/log/log.go (hunk 2 above)
Comment (jcollins-axway) [reply pkg/util/log/log.go:18]: I orig just wanted Set, but the tests needed it to be unset.  In actual usage unset is not needed

File: pkg/util/log/log.go (hunk 1 above)
Comment (jcollins-axway) [reply pkg/util/log/log.go:15]: golang defaults all bools to false, that is why I only wanted the set function

File: pkg/util/log/log.go (hunk 2 above)
Comment (dfeldick) [reply pkg/util/log/log.go:18]: ok

File: pkg/util/log/log.go (hunk 1 above)
Comment (dfeldick) [reply pkg/util/log/log.go:15]: oh, right.

**Review by dfeldick — APPROVED** [review]

**Pull request merged**